Robust_Lab.qmd

```diff
@@ -93,27 +257,112 @@ Use the `stars` data in `robustbase`. This data looks at the relationship betwee
+   generate(reps = 10000, type = "permute") %>%
+   calculate(stat = "diff in means")
+
+```
+
+```{r}
+sample_size <- nrow(my_penguins) # length of dataset
+perm_reps   <- 10000 # number of permutations you want to do
+many.perm <- my_penguins    %>%
+  # this function is in the infer package. What it is doing is creating 
+  rep_sample_n(size = sample_size, replace = FALSE, reps = perm_reps) %>% 
+  mutate(perm_treatment = sample(sex, size = n(), replace = FALSE))  %>%
+  group_by(replicate, perm_treatment)
+many.perm
+
+many.perm.means <- many.perm %>%
+  summarise(mean_group = mean(body_mass_g), .groups = "drop")%>%
+  group_by(replicate)
+many.perm.means
+
+
+many.perm.diffs <- many.perm.means %>%
+  summarise(diff_value = diff(mean_group))
+
+
+
+many.perm.diffs_1 <- many.perm.diffs %>% 
+  mutate(abs_obs_dif = abs(pull(mean_diff)),
+         abs_perm_dif = abs(diff_value),
+         as_or_more_extreme = abs_perm_dif >= abs_obs_dif)
+
+mean(many.perm.diffs_1$as_or_more_extreme)
+
+```
+
 d\. Plot the null-hypothesis distribution (NHD) for the difference
 
+```{r}
+ null_distn %>%
+  visualize() +shade_p_value(obs_stat = df_diff, direction = "two-sided")
+```
+
 e\. Compare the observed mean difference to the NHD (is *p* \< .05?)
 
+```{r}
+#No, it is not.
+
+```
+
 6.  Suppose a replication experiment was conducted to further examine the interaction effect between driving difficulty and conversation difficulty on driving errors in a driving simulator. In the replication, the researchers administered the same three levels of conversation difficulty; (1) control, (2) easy, (3) difficult (C, E, D) but assume that they added a third level of driving difficulty; (1) low, (2) moderate, (3) difficult (L, M, D). Assume the design was completely between subjects and conduct a factorial ANOVA to test the main effects of conversation and driving difficulty as well as the interaction effect. The DV is the number of errors committed in the driving simulator.
 
     ```{r}
     library(tidyverse)
     fac_data<-read_csv("https://raw.githubusercontent.com/jgeller112/psy503-psych_stats/master/static/assignment/data/fact_final.csv")
 
-
     ```
 
     a\. Run a permutation test (ANOVA)
 
+    ```{r}
+    library(permuco)
+    #lmperm() # linear models
+    #aovperm() # anova models
+    #perm.lmer() # lmms
+
+    #Change convo and drive to ordered factors
+    fac_data$convo <- factor(fac_data$convo, ordered = TRUE,
+           levels = c("C", "E", "D"))
+    fac_data$drive <- factor(fac_data$drive, ordered = TRUE,
+           levels = c("L", "M", "D"))
+
+    permuco::aovperm(errors ~ convo * drive,data=fac_data,np=10000)
+
+    ```
+
     b\. How would you follow-up significant effects in this context?
 
+    ```{r}
+    #The interaction is not significant, so there would be no follow-up tests.  However, if the interaction were significant, a simple slopes analysis could be performed to follow up the significant interaction
+    ```
```

Comment:
The main effects are significant.

What I would do is use bootstraps for the mean differences between the conditions/groups.